python/JSONtoObj.c

```diff
@@ -185,40 +187,60 @@ PyObject* JSONToObj(PyObject* self, PyObject *args, PyObject *kwargs)
       return NULL;
   }
 
-  if (PyBytes_Check(arg))
+  Py_buffer buffer;
+  size_t sarg_length;
+  char * raw;
+  bool is_bytes_like = !PyObject_GetBuffer(arg, &buffer, PyBUF_C_CONTIGUOUS);
+  if (is_bytes_like)
   {
-      sarg = arg;
+    #ifdef PYPY_VERSION
+      // PyPy's buffer protocol implementation is buggy: https://foss.heptapod.net/pypy/pypy/-/issues/3872
+      if (!PyBytes_Check(arg) && !PyByteArray_Check(arg)) {
+        PyBuffer_Release(&buffer);
+        PyErr_Format(PyExc_TypeError, "Arbitrary bytes-like objects are not supported on PyPy, Use either string, bytes, or bytearray");
+        return NULL;
+      }
+    #endif
+    raw = buffer.buf;
+    sarg_length = buffer.len;
   }
-  else
-  if (PyUnicode_Check(arg))
-  {
-    sarg = PyUnicode_AsEncodedString(arg, NULL, "surrogatepass");
-    if (sarg == NULL)
+  else {
```

Comment:
Nit: the bracket's on a separate line in almost the entire code.

Reply:
Grr, well spotted. How I hate that convention - such a waste of a line.